llvm/lib/Target/SPIRV/SPIRVModuleAnalysis.cpp

```diff
@@ -76,18 +76,23 @@ getSymbolicOperandRequirements(SPIRV::OperandCategory::OperandCategory Category,
                                SPIRV::RequirementHandler &Reqs) {
   static AvoidCapabilitiesSet
       AvoidCaps; // contains capabilities to avoid if there is another option
-  unsigned ReqMinVer = getSymbolicOperandMinVersion(Category, i);
-  unsigned ReqMaxVer = getSymbolicOperandMaxVersion(Category, i);
-  unsigned TargetVer = ST.getSPIRVVersion();
-  bool MinVerOK = !ReqMinVer || !TargetVer || TargetVer >= ReqMinVer;
-  bool MaxVerOK = !ReqMaxVer || !TargetVer || TargetVer <= ReqMaxVer;
+
+  unsigned ReqMinVerDec = getSymbolicOperandMinVersion(Category, i);
+  VersionTuple ReqMinVer = VersionTuple(ReqMinVerDec / 10, ReqMinVerDec % 10);
+  unsigned ReqMaxVerDec = getSymbolicOperandMaxVersion(Category, i);
+  VersionTuple ReqMaxVer = VersionTuple(ReqMaxVerDec / 10, ReqMaxVerDec % 10);
+  VersionTuple SPIRVVersion = ST.getSPIRVVersion();
```

Comment:
Seems like the `getSymbolicOperandMinVersion` and `getSymbolicOperandMaxVersion` are only used here.
Shall they be updated to return the VersionTuple directly, and hide the `/10` and `%10` in it?

Reply:
Thanks for catching this!